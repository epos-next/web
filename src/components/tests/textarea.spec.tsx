import Textarea from "@components/textarea";
import { render } from "@testing-library/react";
import React from "react"

it("should match snapshot", () => {
    const { asFragment } = render(<Textarea/>)
    expect(asFragment()).toMatchSnapshot()
});

it("should use error state", () => {
    const { asFragment } = render(<Textarea error={true} />)
    expect(asFragment()).toMatchSnapshot()
});
