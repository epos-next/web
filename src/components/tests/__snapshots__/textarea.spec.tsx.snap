// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  border: 2px solid var(--light-primary);
  border-radius: 10px;
  outline: none;
  padding: 14px;
  font-size: 18px;
  color: var(--primary);
  -webkit-transition: 150ms border;
  transition: 150ms border;
}

.c0::-webkit-input-placeholder {
  color: var(--light-primary);
}

.c0::-moz-placeholder {
  color: var(--light-primary);
}

.c0:-ms-input-placeholder {
  color: var(--light-primary);
}

.c0::placeholder {
  color: var(--light-primary);
}

.c0:focus {
  border: 2px solid var(--contrast);
}

.c0[data-error=true] {
  color: var(--error);
  border: 2px solid var(--error);
}

.c0[data-error=true]:focus {
  border: 2px solid var(--error);
}

@media screen and (max-width:768px) {
  .c0 {
    padding: 13px;
  }
}

<textarea
    class="c0"
  />
</DocumentFragment>
`;

exports[`should use error state 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  border: 2px solid var(--light-primary);
  border-radius: 10px;
  outline: none;
  padding: 14px;
  font-size: 18px;
  color: var(--primary);
  -webkit-transition: 150ms border;
  transition: 150ms border;
}

.c0::-webkit-input-placeholder {
  color: var(--light-primary);
}

.c0::-moz-placeholder {
  color: var(--light-primary);
}

.c0:-ms-input-placeholder {
  color: var(--light-primary);
}

.c0::placeholder {
  color: var(--light-primary);
}

.c0:focus {
  border: 2px solid var(--contrast);
}

.c0[data-error=true] {
  color: var(--error);
  border: 2px solid var(--error);
}

.c0[data-error=true]:focus {
  border: 2px solid var(--error);
}

@media screen and (max-width:768px) {
  .c0 {
    padding: 13px;
  }
}

<textarea
    class="c0"
    data-error="true"
  />
</DocumentFragment>
`;
